src/main/java/ui/UI.java

```diff
@@ -540,4 +541,27 @@ public boolean isWindowMinimized() {
     public boolean isWindowFocused() {
         return mainStage.isFocused();
     }
+
+    /**
+     * Warns user if the Java runtime version is lower than HT's requirement
+     */
+    private void warnIfJavaVersionOutdated() {
+        try {
+            JavaVersion javaRuntimeVersion = JavaVersion.fromString(System.getProperty("java.runtime.version"));
+            JavaVersion javaRequiredVersion = JavaVersion.fromString(JavaVersion.REQUIRED_VERSION);
```

Comment:
Seems like the required version should be a property of the application (UI) instead.

This is also a bit verbose. Maybe we could drop some of the prefixes for the local variables (which are less likely to be ambiguous)?